src/language/customPathInstaller.ts

```diff
@@ -0,0 +1,152 @@
+import { promisify } from 'util';
+import * as os from 'os';
+import * as fs from 'fs';
+import { ExtensionContext, OutputChannel, window } from 'vscode';
+
+const existsAsync = promisify(fs.exists);
+import * as path from 'path';
+import { ConfigurationConstants, LanguageServerConstants } from '../constants';
+import Configuration from '../configuration';
+import { getPreferredVersion } from './dafnyInstallation';
+import { Executable } from 'vscode-languageclient/node';
+import { getDotnetExecutablePath } from '../dotnet';
+
+const mkdirAsync = promisify(fs.mkdir);
+
+export class CustomPathInstaller {
+  public constructor(
+    public readonly context: ExtensionContext,
+    public readonly statusOutput: OutputChannel
+  ) {}
+
+  // We cache the cli path so that we don't need to copy it every time.
+  private cliPath: string | undefined | null = null;
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  public async getExecutable(context: ExtensionContext, server: boolean, newArgs: string[], oldArgs: string[]): Promise<Executable | undefined> {
+    const configuredCliPath = await this.getCliPath(context);
+    if(!configuredCliPath) {
+      return undefined;
+    }
+
+    if(server) {
+      newArgs.unshift('server');
+    }
+    if(configuredCliPath.endsWith('.dll')) {
+      const { path: dotnetExecutable } = await getDotnetExecutablePath();
+      return {
+        command: dotnetExecutable,
+        args: [ configuredCliPath ].concat(newArgs)
```

Comment:
I thought that the newArgs for the DLL were different than the newArgs for the dafny server command. How come you can use them here?

Reply:
Both are using the Dafny server command, but one is using a standalone executable and the other a .dll that is run using dotnet